webkit-extension/MailPage.vala

```diff
@@ -0,0 +1,124 @@
+                    page.get_dom_document ().get_document_element ().query_selector ("#message-body").set_inner_html (body);
+                    return true;
+                } catch (Error e) {
+                    warning ("Unable to set message body content: %s", e.message);
+                }
+
+                return false;
+            case "get-body-html":
+                try {
+                    var body_html = page.get_dom_document ().get_document_element ().query_selector ("body").get_inner_html ();
+                    message.send_reply (new WebKit.UserMessage ("get-body-html", new Variant.take_string ((owned) body_html)));
+                    return true;
+                } catch (Error e) {
+                    warning ("Unable to get message body content: %s", e.message);
+                }
+
+                return false;
+            case "get-page-height":
+                var height = (int32)page.get_dom_document ().get_document_element ().get_offset_height ();
+                message.send_reply (new WebKit.UserMessage ("get-page-height", new Variant.int32 (height)));
+                return true;
+            case "set-image-loading-enabled":
+                var enabled = message.parameters.get_boolean ();
+                show_images = enabled;
+                if (enabled) {
+                    var images = page.get_dom_document ().get_images ();
+                    for (int i = 0; i < images.length; i++) {
+                        var image = (WebKit.DOM.HTMLImageElement)images.item (i);
+                        image.set_src (image.get_src ());
+                    }
+                }
+
+                return true;
+            case "execute-editor-command":
+                string command, argument;
+                message.parameters.get ("(ss)", out command, out argument);
+                var document = page.get_dom_document ();
+                document.exec_command (command, false, argument);
+                return true;
+            case "query-command-state":
+                unowned string command = message.parameters.get_string ();
+                var document = page.get_dom_document ();
+                var state = document.query_command_state (command);
+                message.send_reply (new WebKit.UserMessage ("query-command-state", new Variant.boolean (state)));
+                return true;
+            case "get-selected-text":
+                try {
+                    var selection_range = page.get_dom_document ().default_view.get_selection ().get_range_at (0);
+                    if (selection_range != null) {
+                        message.send_reply (new WebKit.UserMessage ("get-selected-text", new Variant.string (selection_range.text)));
+                        return true;
+                    } else {
+                        GLib.message ("no selection range");
+                    }
+                } catch (Error e) {
+                    GLib.message (e.message);
```

Comment:
Any reason why you're using `message` here and on line 93 instead of `debug` and `warning`?

Reply:
I don't really know why, I've changed their level